### PR TITLE
WhiteListing investors

### PR DESCRIPTION
**Pull request by Janther**
Adding:
  - whitelisting
  - individual hardcap
  - enforcing the use of Prettier for js formatting on tests.

### CODE REVIEW

**Review by matt-lough — APPROVED** [review]

File: package.json [hunk 1]
```diff
@@ -22,5 +22,8 @@
     "chai": "^4.1.1",
     "ethereumjs-abi": "^0.6.4",
     "truffle": "^3.4.9"
+  },
+  "devDependencies": {
+    "prettier": "^1.6.1"
```
Comment (IsaacPayne) [inline package.json:27]: 👍

**Review by IsaacPayne — APPROVED** [review]

File: contracts/Contribution.sol [hunk 2]
```diff
@@ -200,11 +218,25 @@ contract Contribution is Controlled, TokenController {
...
+  /// @return Total tokens available for the sale in weis.
   function tokensForSale() public constant returns(uint256) {
     return totalSupplyCap > totalSold ? totalSupplyCap.sub(totalSold) : 0;
   }
 
+  /// @return Total tokens available for a specific address in weis.
+  function tokensForSale(address investor) public constant returns(uint256) {
+    uint256 cap;
+    uint256 sold;
+    if (getBlockTimestamp() <= startTime + 1 days) {
```
Comment (rstormsf) [inline contracts/Contribution.sol]: I don't understand why there is `+ 1 days`

File: contracts/Contribution.sol [hunk 3]
```diff
@@ -16,6 +16,10 @@ contract Contribution is Controlled, TokenController {
   uint256 public totalSupplyCap;            // Total AIX supply to be generated
   uint256 public totalSold;                 // How much tokens have been sold
 
+  uint256 public numWhitelistedInvestors;
+  mapping (address => bool) public canPurchase;
+  mapping (address => uint256) public individualSold;
```
Comment (rstormsf) [inline contracts/Contribution.sol:21]: I thought there could be something like:
```
  struct WhitelistedInvestor {
      uint256 canPurchase;
      uint256 individualSold;
  }
  mapping(address => WhitelistedInvestor) investors;
```

File: contracts/Contribution.sol [hunk 4]
```diff
@@ -96,6 +100,18 @@ contract Contribution is Controlled, TokenController {
     Initialized(initializedBlock);
   }
 
+  function whitelist(address investor) {
+    if (canPurchase[investor]) return;
+    numWhitelistedInvestors++;
+    canPurchase[investor] = true;
+  }
+
```
Comment (rstormsf) [inline contracts/Contribution.sol]: so there is no `isWhitelisted()` method to handle the `doBuy` ?

File: contracts/Contribution.sol (hunk 2 above)
Comment (Janther) [reply contracts/Contribution.sol]: good catch, this is meant to be a placeholder for the X hours after (I'm not sure about adding a variable here and all the logic to initiate it)

File: contracts/Contribution.sol (hunk 3 above)
Comment (Janther) [reply contracts/Contribution.sol:21]: I feel like these are 2 completely different and kind of unrelated values.
I don't see any benefit of creating a structure every time.

Still is a start

**Pull request merged**